### PR TITLE
Improve isManagedWorkload

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
This is adding check node name

### CODE REVIEW

File: pkg/auth/rbac.go [hunk 1]
```diff
@@ -67,10 +68,11 @@ type rbacConnection struct {
...
 }
 
-func NewRbac(workloadObj *bpf.BpfKmeshWorkload) *Rbac {
+func NewRbac(workloadObj *bpf.BpfKmeshWorkload, workloadCache cache.WorkloadCache) *Rbac {
 	return &Rbac{
-		policyStore: newPolicystore(),
-		bpfWorkload: workloadObj,
+		policyStore:   newPolicystore(),
+		workloadCache: workloadCache,
+		bpfWorkload:   workloadObj,
```
Comment (supercharge-xsy) [inline pkg/auth/rbac.go:75]: Why add workload info to the rbac cache？

Comment (hzxuzhonghu) [reply pkg/auth/rbac.go:75]: because this patch is also removing the global cache var

File: pkg/controller/workload/cache/workload_cache.go [hunk 2]
```diff
@@ -25,35 +25,38 @@ import (
...
-	return &workloadStore{
+func NewWorkloadCache() *cache {
+	return &cache{
 		byUid:  make(map[string]*workloadapi.Workload),
 		byAddr: make(map[NetworkAddress]*workloadapi.Workload),
 	}
 }
 
-func (w *workloadStore) GetWorkloadByUid(uid string) *workloadapi.Workload {
+func (w *cache) GetWorkloadByUid(uid string) *workloadapi.Workload {
```
Comment (LiZhenCheng9527) [inline pkg/controller/workload/cache/workload_cache.go:53]: Wouldn't it be better to modify the `w` as well.

Comment (hzxuzhonghu) [reply pkg/controller/workload/cache/workload_cache.go:53]: right, should do

File: pkg/controller/workload/workload_processor.go [hunk 3]
```diff
@@ -467,16 +470,26 @@ func (p *Processor) handleDataWithoutService(workload *workloadapi.Workload) err
...
+		oldIdentity := p.getIdentityByUid(workload.Uid)
+		newIdentity := spiffe.Identity{
+			TrustDomain:    workload.TrustDomain,
+			Namespace:      workload.Namespace,
+			ServiceAccount: workload.ServiceAccount,
+		}.String()
+		if oldIdentity == "" {
+			// This is the case workload added first time
+			p.Sm.SendCertRequest(newIdentity, kmeshsecurity.ADD)
+		} else if oldIdentity != newIdentity {
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/workload_processor.go:484]: maybe no need to check this, in real case workload identity cannot change unless recreate a pod

**Comment by hzxuzhonghu** [comment]
@supercharge-xsy any other comments

**Pull request merged**